### PR TITLE
sensor drivers: initialization cleanup

### PR DESCRIPTION
**Pull request by OlegHahm**
Addressing #5156.

### CODE REVIEW

**Comment by OlegHahm** [comment]
@haukepetersen, please take a look if you're okay with this before I adopt this for other sensors.

File: drivers/l3g4200d/l3g4200d.c [hunk 1]
```diff
@@ -33,20 +33,20 @@
 
 #define MAX_VAL         0x7fff
 
-int l3g4200d_init(l3g4200d_t *dev, i2c_t i2c, uint8_t address,
-                  gpio_t int1_pin, gpio_t int2_pin,
-                  l3g4200d_mode_t mode, l3g4200d_scale_t scale)
+int l3g4200d_init(l3g4200d_t *dev, const l3g4200d_params_t *params)
 {
     uint8_t tmp;
+    l3g4200d_t *dp = (l3g4200d_t*) params;
```
Comment (aabadie) [inline drivers/l3g4200d/l3g4200d.c]: It should be `params->dev`, right ?

**Comment by haukepetersen** [comment]
NACK, this does not work. The important thing here is, that the `xxx_params_t` struct typically resides in the ROM (as static configuration data), while the dev descriptor `xxx_t` typically has to be in RAM, as it mostly contains dynamic fields. The trade-off a driver developer has to asses then is, if it makes sense to copy all parameters from ROM to RAM, or if it makes sense to only copy a sub-set..

**Comment by OlegHahm** [comment]
I see that this PR needs serious rework. So instead of keeping it open for a too long time, I will close it and open a new one once I find the time for the rework.

**Pull request closed without merging**